### PR TITLE
Bug 1132936 - Create unit tests for IdentityPopup class in toolbars.py

### PR DESCRIPTION
**Pull request by galgeek**
https://bugzilla.mozilla.org/show_bug.cgi?id=1132936

This PR replaces #86. Thanks @heisenberg!

### CODE REVIEW

**Comment by galgeek** [comment]
Hi, @whimboo and @chmanchester!

The appveyor build looks fine for the tests themselves, but maybe the uninstall is failing, or the syntax error reported at the end is considered a failure?

**Comment by chmanchester** [comment]
We're looking into this in https://bugzilla.mozilla.org/show_bug.cgi?id=1125732

File: firefox_puppeteer/tests/test_toolbars.py [hunk 1]
```diff
@@ -135,3 +137,57 @@ def test_matching_text(self):
...
             for match_fragment in all_matches:
                 self.assertIn(match_fragment, (input_text, input_text.upper()))
+
+
+class TestIdentityPopup(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+        self.identity_popup = self.browser.navbar.locationbar.identity_popup
+
```
Comment (whimboo) [inline firefox_puppeteer/tests/test_toolbars.py]: Don't forget to add the tearDown, which force closes the popup. I assume you will add it once the close() method has been landed via the other PR.

File: firefox_puppeteer/tests/test_toolbars.py [hunk 2]
```diff
@@ -135,3 +137,57 @@ def test_matching_text(self):
...
+    def test_identity_popup_elements(self):
+        self.assertEqual('box', self.identity_popup.box.get_attribute('localName'))
```
Comment (whimboo) [inline firefox_puppeteer/tests/test_toolbars.py]: The expected value you are comparing with should be the second argument everywhere.

File: firefox_puppeteer/tests/test_toolbars.py [hunk 3]
```diff
@@ -135,3 +137,57 @@ def test_matching_text(self):
...
+                         self.identity_popup.owner_location.get_attribute('localName'))
+        self.assertEqual('vbox', self.identity_popup.permissions.get_attribute('localName'))
+        self.assertEqual('description', self.identity_popup.verifier.get_attribute('localName'))
+
+        self.marionette.execute_script("""
+          arguments[0].hidePopup();
+        """, script_args=[self.identity_popup.popup])
+
+    @skip_under_xvfb
+    def test_is_open(self):
```
Comment (whimboo) [inline firefox_puppeteer/tests/test_toolbars.py]: Once the close() method exists, we can name it 'test_open_close'.

File: firefox_puppeteer/tests/test_toolbars.py [hunk 4]
```diff
@@ -135,3 +137,57 @@ def test_matching_text(self):
...
+        data_uri = 'data:text/html,<title>Title</title>'
+        locationbar = self.browser.navbar.locationbar
+        locationbar.load_url(data_uri)
+
+        self.assertTrue(not self.identity_popup.is_open)
```
Comment (whimboo) [inline firefox_puppeteer/tests/test_toolbars.py]: Therefore we have self.assertFalse.

File: firefox_puppeteer/tests/test_toolbars.py [hunk 5]
```diff
@@ -135,3 +137,57 @@ def test_matching_text(self):
...
+
+        self.identity_popup.box.click()
+        self.wait_for_condition(lambda _: self.identity_popup.is_open)
+
+        self.marionette.execute_script("""
+          arguments[0].hidePopup();
+        """, script_args=[self.identity_popup.popup])
```
Comment (whimboo) [inline firefox_puppeteer/tests/test_toolbars.py]: Add another assert which checks that it is closed now.

File: firefox_puppeteer/tests/test_toolbars.py [hunk 6]
```diff
@@ -135,3 +135,77 @@ def test_matching_text(self):
...
+        FirefoxTestCase.setUp(self)
+        self.identity_popup = self.browser.navbar.locationbar.identity_popup
+
+    def tearDown(self):
+        try:
+            self.identity_popup.close(force=True)
+        except NoSuchElementException:
+            # TODO: A NoSuchElementException may be thrown here when tests accessing the
+            # identity_popup.popup element are skipped.
+            pass
```
Comment (whimboo) [inline firefox_puppeteer/tests/test_toolbars.py]: Ok, so this is in parity with the Autocomplete popup unit tests. We have to get a bug filed on it. Interestingly `close()` can be called in setUp without this failure. I think we have a wrong window handle here or the focus is somewhere else. @galgeek can you please file this bug?

**Comment by whimboo** [comment]
PR has been merged as https://github.com/mozilla/firefox-ui-tests/commit/9505b0e96c5cae85814862d959aa1e3631b31ff0. Thanks!

**Pull request closed without merging**